ydb/core/viewer/viewer_cluster.h

```diff
@@ -462,23 +466,40 @@ class TJsonCluster : public TViewerPipeClient {
             }
         }
 
+        struct TMemoryStats {
+            ui64 Total = 0;
+            ui64 Limit = 0;
+        };
+
+        std::unordered_set<TString> hostPassed;
+        std::unordered_map<TString, TMemoryStats> memoryStats;
+
         for (TNode& node : NodeData) {
             const NKikimrWhiteboard::TSystemStateInfo& systemState = node.SystemState;
             (*ClusterInfo.MutableMapDataCenters())[node.DataCenter]++;
-            if (systemState.HasNumberOfCpus()) {
+            if (hostPassed.insert(systemState.GetHost()).second) {
```

Comment:
what about multiple nodes on the same host?

Reply:
> what about multiple nodes on the same host?

that's exactly what this fix made for!